### PR TITLE
Strategy Pattern for PyMC-Marketing

### PR DESCRIPTION
**Pull request by wd60622**

### CODE REVIEW

**Comment by wd60622** [comment]
Hey guys,
I thought using a code design pattern was very helpful for our PR so I'm writing up this personal blog post to talk about it. 
If you have time and would like to take a look, would love to hear your thoughts! I will likely merge in next few days then make a separate LinkedIn post down the line. Really want to boast about the 3 * 6 * 2 new out-of-box models. So epic

@juanitorduz, I can tag link you directly if you'd like. I would like to give you credit for all your hard work in the review!

CC: @cetagostini

File: docs/blog/posts/2024/pymc-marketing-strategy-pattern.md [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+    "tanh": create_tanh_saturation(b=1.25, c=0.5),
+    "Hill": create_hill_saturation(sigma=1.5, beta=5.5, lam=0.75),
+    "Michaelis-Menten": create_michaelis_menten_saturation(lam=0.1, alpha=1.1),
+    "Infinite Returns": create_infinite_returns(beta=1.25),
+}
+
+# Application
+x = np.linspace(0, 1, 100)
+ax = plt.subplot(111)
+for name, saturation in saturation_functions.items():
```
Comment (juanitorduz) [inline docs/blog/posts/2024/pymc-marketing-strategy-pattern.md:259]: Add an empty line before for loop?

File: docs/blog/posts/2024/pymc-marketing-strategy-pattern.md [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+Though there is a little more boilerplate, the two step process is still used.
+
+1. Creation
+1. Application
+
+```python
+saturation = InfiniteReturns()
+
+x = np.linspace(0, 1, 100)
```
Comment (juanitorduz) [inline docs/blog/posts/2024/pymc-marketing-strategy-pattern.md:309]: Add  an empty line before` with Model`?

**Review by juanitorduz — COMMENTED** [review]
Fantastic blog! I love the new design (I learned a lot during the reviews :D )

**Comment by wd60622** [comment]
Thanks for taking a look @juanitorduz !
Appreciate the feedback

**Pull request merged**